internal/subscriber/db.go

```diff
@@ -0,0 +1,357 @@
+/*******************************************************************************
+ * Copyright 2022 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package subscriber
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/arangodb/go-driver"
+	"github.com/arangodb/go-driver/http"
+	sdkContract "github.com/project-alvarium/alvarium-sdk-go/pkg/contracts"
+	"github.com/project-alvarium/alvarium-sdk-go/pkg/message"
+	logInterface "github.com/project-alvarium/provider-logging/pkg/interfaces"
+	"github.com/project-alvarium/provider-logging/pkg/logging"
+	"github.com/project-alvarium/scoring-apps-go/internal/config"
+	"github.com/project-alvarium/scoring-apps-go/pkg/documents"
+	"sync"
+	"time"
+)
+
+type arangoClient struct {
+	cfg    config.ArangoConfig
+	chPub  chan string
+	chSub  chan message.SubscribeWrapper
+	client driver.Client
+	logger logInterface.Logger
+}
+
+func NewArangoClient(sub chan message.SubscribeWrapper, pub chan string, dbConfig config.DatabaseInfo, logger logInterface.Logger) (arangoClient, error) {
+	cfg, ok := dbConfig.Config.(config.ArangoConfig)
+	if !ok {
+		return arangoClient{}, fmt.Errorf("invalid config type, expected %s", config.DBArango)
+	}
+	c := arangoClient{
+		cfg:    cfg,
+		chPub:  pub,
+		chSub:  sub,
+		logger: logger,
+	}
+
+	conn, err := http.NewConnection(
+		http.ConnectionConfig{
+			Endpoints: []string{cfg.Provider.Uri()},
+		})
+	if err != nil {
+		return arangoClient{}, err
+	}
+	client, err := driver.NewClient(
+		driver.ClientConfig{
+			Connection: conn,
+		})
+	if err != nil {
+		return arangoClient{}, err
+	}
+	c.client = client
+	return c, nil
+}
+
+func (c *arangoClient) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup) bool {
+	err := c.initGraph(ctx)
+	if err != nil {
+		c.logger.Error(err.Error())
+		return false
+	}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		for {
+			item, ok := <-c.chSub
+			if ok {
+				switch item.Action {
+				case message.ActionCreate:
+					c.logger.Write(logging.DebugLevel, "handling create")
+					err = c.handleCreateTransit(ctx, item.Content)
+				case message.ActionTransit:
+					c.logger.Write(logging.DebugLevel, "handling transit")
+					err = c.handleCreateTransit(ctx, item.Content)
+				case message.ActionMutate:
+					c.logger.Write(logging.DebugLevel, "handling mutate")
+					err = c.handleMutate(ctx, item.Content)
+				default:
```

Comment:
Should we implement `publish` here?
`case message.ActionPublish`

Reply:
I looked into this and it is a gap. Because it will require some additional work, we should handle as a follow-up issue.